File: src/main/java/com/dumbdogdiner/warrior/api/kit/abilities/MedicAbility.java
package com.dumbdogdiner.warrior.api.kit.abilities;

import com.dumbdogdiner.warrior.api.user.WarriorUser;
import com.dumbdogdiner.warrior.api.kit.Ability;
import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

public class MedicAbility extends Ability {

    private static final String[] desc = {
            "&7Fully &frestores your health &7and",
            "&7removes all status effects."
    };

    public MedicAbility(String name, int cost, int minStreak) {
        super(name, cost, minStreak, true, desc);
    }


    @Override
    public Runnable run(WarriorUser user) {
        Player p = user.getBukkitPlayer();

        return () -> {
            for(PotionEffect potion : p.getActivePotionEffects()) {
                p.removePotionEffect(potion.getType());
            }
            p.addPotionEffect(new PotionEffect(PotionEffectType.REGENERATION, 100, 1));
        };
    }
}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/kits/abilities/MedicAbility.kt
package com.dumbdogdiner.warrior.kits.abilities

import com.dumbdogdiner.warrior.api.kit.WarriorAbility
import com.dumbdogdiner.warrior.api.user.WarriorUser
import org.bukkit.Sound
import org.bukkit.attribute.Attribute

class MedicAbility : WarriorAbility {

    override fun getName(): String {
        return "Medic"
    }

    override fun run(user: WarriorUser<*>) {
        val player = user.toBukkit()
        val maxHealth = player.getAttribute(Attribute.GENERIC_MAX_HEALTH)!!.baseValue

        if(this.shouldExecute(user)) {
            // this makes me cum tbh
            player.health = maxHealth
            player.playSound(player.location, Sound.BLOCK_ENCHANTMENT_TABLE_USE, 1f, 1f)
        }
    }

    override fun shouldExecute(user: WarriorUser<*>): Boolean {
        val maxHealth = user.toBukkit().getAttribute(Attribute.GENERIC_MAX_HEALTH)!!.baseValue
        return user.toBukkit().health < maxHealth
    }

}